src/App.tsx

```diff
@@ -1,20 +1,40 @@
-import React from "react";
+import React, { useEffect} from "react";
 import { Routes, Route } from "react-router-dom";
 import Home from "./pages/home/Home";
 import Profile from "./pages/profile/Profile";
 import Feeds from "./pages/feed/Feeds";
-import Navbar from "./shared/navbar/Navbar";
+import { Signup } from "./pages/authentication/signup/Signup";
+import { Login } from "./pages/authentication/login/Login";
+import { PrivateRoute } from "./shared/privateRoute/PrivateRoute";
+import { ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { getCurrentUser } from "./features/auth/authSlice";
+import { useAppDispatch } from "./app/hooks";
+
 
 function App() {
+
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    dispatch(getCurrentUser());
+  }, [dispatch]);
```

Comment:
fix: I am not sure if you should pass the dispatch fucntion here, because in this every time there is some change in redux state the entire app would rerender